[DOC] Add Streaming doc for Tempo data source (#91432)
grafana/grafana: docs/sources/datasources/tempo/configure-tempo-data-source.md

# Configure the Tempo data source

> The Tempo data source sets how your Tempo data connects to Grafana and lets you configure features and integrations with other telemetry signals.

Reviewer: I tend to think about it the other way, a data source config sets how Grafana connects with your database.

I think of the query path like:

Dashboard -> Panel -> Query -> Data source -> Database

```suggestion
The Tempo data source sets how Grafana connects to your Tempo database and lets you configure features and integrations with other telemetry signals.
```

{{< youtube id="cqHO0oYW6Ic" start="298" >}}

## Streaming

<!-- The traceQLStreaming toggle will be deprecated in Grafana 11.2 and removed in 11.3. -->

Streaming enables TraceQL query results to be displayed as they become available. Without streaming, no results are displayed until all results have returned.

{{< docs/public-preview product="TraceQL streaming results" >}}

### Requirements

To use streaming, you need to:

- Be running Tempo version 2.2 or newer, or Grafana Enterprise Traces (GET) version 2.2 or newer, or be using Grafana Cloud Traces.
- For self-managed Tempo or GET instances: If your Tempo or GET instance is behind a load balancer or proxy that doesn't supporting gRPC or HTTP2, streaming may not work and should be disabled.

### Activate streaming

For streaming to work for a particular Tempo data source, set your Grafana's `traceQLStreaming` [feature toggle](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/feature-toggles/) to true and set **Streaming** to enabled in your Tempo data source configuration.

> 

Author: ```suggestion

![Streaming section in Tempo data source](/media/docs/grafana/data-sources/tempo-data-source-streaming-v11.2.png)

```

{{< youtube id="cqHO0oYW6Ic" start="298" >}}

## Streaming

<!-- Streaming toggle will be deprecated in Grafana 11.2 and removed in 11.3. -->

> Streaming provides immediate visibility of incoming traces on the results table for a TraceQL query.

Author: It's clearer in how it works, but we can structure it better. The first sentence is what we use in the Streaming section of the docs. What about this?
```suggestion
Streaming enables TraceQL query results to be displayed as they become available. Without streaming, you no results are displayed until all results have returned. 
```

The Streaming section is available in the Tempo data source settings for Grafana Cloud.
This capability requires Tempo 2.2 and onwards.

### Requirements

To use this feature, you need to:

> - Use Tempo version 2.2 or newer. Grafana Cloud uses has the latest version of Tempo.

Author: Do we need to mention streaming in the GET docs? I don't think we mention much of anything like TraceQL.

- For self-managed Tempo instances: If your Tempo instance is behind a load balancer or proxy that doesn't supporting gRPC or HTTP2, streaming may not work and should be disabled.

For streaming to work for a particular Tempo data source, set both `traceQLStreaming` feature toggle to true and **Streaming** enabled in the the data source to allow streaming.

If the data source allows streaming but `traceQLStreaming=false`, no streaming will occur.

If the data source has streaming disabled and `traceQLStreaming=true`, no streaming wil happen for that data source.

> ### Activate streaming

Reviewer 2: This section seems sort of redundant when compared to what is already above so perhaps not needed? Will defer to you however!

Author: I think I might restructure the information here. Some of the content we have in Requirements should be in the procedural part.

Author: Restructured.

To activate the Streaming section in your Tempo data source settings:

1. Activate the `traceQLStreaming` feature toggle in your Grafana configuration file.

- If you are using Grafana Cloud, contact Customer Support to activate this feature toggle.

> 1. Select the **Queries** toggle to activate streaming for queries.

Reviewer 2: ```suggestion
1. Select the **Queries** toggle on your Tempo data source settings page.
```

> 

Reviewer 2: Do you think it would be helpful to put a screenshot? How about this one? 
<img width="661" alt="Screenshot 2024-08-01 at 10 06 43 PM" src="https://github.com/user-attachments/assets/59e304ed-d05d-4618-9d71-1c9648ab1f37">

Author: Thank you. I'll add one. I may also add ones for the other secctions as well.

Author: Done.
